### PR TITLE
docs: add documentation for the argocd cluster commands

### PR DESCRIPTION
**Pull request by morremeyer**
This adds initial non-reference documentation for the `argocd cluster` commands as part of #14531.

The main goal here is to bootstrap that page and point users wanting to remove the `in-cluster` cluster to
the right setting.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] ~I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.~ not applicable
* [x] ~I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.~ not applicable
* [x] ~Does this PR require documentation updates?~ not applicable
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] ~I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.~ not applicable~
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [x] ~My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.~ not applicable
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.

### CODE REVIEW

File: docs/operator-manual/cluster-management.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Cluster Management
+
+This guide is for operators looking to manage clusters on the CLI. If you want to use Kubernetes resources for this, check out [Declarative Setup](./declarative-setup.md#clusters).
+
+Not all commands are described here, see the [argocd cluster Command Reference](../user-guide/commands/argocd_cluster.md) for all available commands.
+
+## Adding a cluster
+
+Run `argocd cluster add $CONTEXT`.
```
Comment (chengfang) [inline docs/operator-manual/cluster-management.md:9]: `$CONTEXT` gives the impression that this env has to be set beforehand. I feel using `<context>` or `context-name` is a bit easier to understand.

File: docs/operator-manual/cluster-management.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+This will connect to the cluster and install the necessary resources for ArgoCD to connect to it.
+Note that you will need privileged access to the cluster.
+
+## Removing a cluster
+
+Run `argocd cluster rm $NAME`.
+
+This removes the cluster with the specified name.
+
+!!!note "in-cluster cannot be removed"
```
Comment (chengfang) [inline docs/operator-manual/cluster-management.md:22]: I believe a space after !!! is needed.

File: docs/operator-manual/cluster-management.md (hunk 1 above)
Comment (morremeyer) [reply docs/operator-manual/cluster-management.md:9]: That makes sense, updated it.

File: docs/operator-manual/cluster-management.md (hunk 2 above)
Comment (morremeyer) [reply docs/operator-manual/cluster-management.md:22]: I checked with the other documentation files, none of them have a space between `!!!` and `note` (or `warning`).

Comment (chengfang) [reply docs/operator-manual/cluster-management.md:22]: ok, good to know that.

**Review by chengfang — APPROVED** [review]
LGTM.

**Review by todaywasawesome — APPROVED** [review]
Seems reasonable. I was debating duplicating whats in the CLI reference but I understand he purpose of the page.

**Pull request merged**